lib/core/src/Cardano/Wallet/Primitive/AddressDerivation/Random.hs

```diff
@@ -128,10 +150,189 @@ hashSeed :: ScrubbedBytes -> ScrubbedBytes
   where
     serialize = BA.convert . cbor . BA.convert
-    cbor = BL.toStrict . CBOR.toLazyByteString . CBOR.encodeBytes
+    cbor = CBOR.toStrictByteString . CBOR.encodeBytes
 
 -- hashSeedForPaperWallet :: ScrubbedBytes -> ScrubbedBytes
 -- hashSeedForPaperWallet = blake2b256
 
 blake2b256 :: ScrubbedBytes -> ScrubbedBytes
 blake2b256 = BA.convert . hash @ScrubbedBytes @Blake2b_256
+
+{-----------------------------------------------------------------------------
+                         Derivation path serialization
+
+In the composition of a Cardano address, the following functions concern the
+"Derivation Path" box.
+
++--------------------------------------------------------------------------------+
+|     Address Root    |     Address Attributes    |           AddrType           |
+|                     |                           |                              |
+|   Hash (224 bits)   |  Der. Path* + Stake + NM  |  PubKey | (Script) | Redeem  |
+|                     |    (open for extension)   |     (open for extension)     |
++--------------------------------------------------------------------------------+
+             |                 |
+             |                 |     +----------------------------------+
+             v                 |     |        Derivation Path           |
++---------------------------+  |---->|                                  |
+|  SHA3-256 >> Blake2b_224  |  |     | ChaChaPoly* AccountIx/AddressIx  |
+|                           |  |     +----------------------------------+
+|  -AddrType                |  |
+|  -ASD* (~AddrType+PubKey) |  |     +----------------------------------+
+|  -Address Attributes      |  |     |       Stake Distribution         |
++---------------------------+  |     |                                  |
+                               |---->|  BootstrapEra | (Single | Multi) |
+                               |     +----------------------------------+
+                               |
+                               |     +----------------------------------+
+                               |     |          Network Magic           |
+                               |---->|                                  |
+                                     | Addr Discr: MainNet vs TestNet   |
+                                     +----------------------------------+
+
+------------------------------------------------------------------------------}
+
+-- | Encode the derivation path (account index and address index) part of a
+-- random HD address.
+--
+-- NOTE: The caller must ensure adequate key length.
+encodeDerivationPath
+    :: Key 'RootK XPub
+    -> Index 'Hardened 'AccountK
+    -> Index 'Soft 'AddressK
+    -> CBOR.Encoding
+encodeDerivationPath rootKey accIx addrIx = mconcat
+    [ CBOR.encodeListLen 3
+    , CBOR.encodeBytes "fuzz" -- fixme: what are these bytes?
+    , CBOR.encodeMapLen 1
+    , CBOR.encodeWord8 0 -- fixme: what is this?
```

Comment:
This is probably the tag that identifies a constructor in the address attribute encoding (That's how cardano-sl used to serialized address attributes. There are in _theory_ several sort of attributes, but in practice, not all of them are used. Constructors are tagged with a particular value to be deserialized to Haskell values.

Beside, if this indeed refers to the address derivation path attribute, it should be `1`